granulate_utils/linux/cgroups/memory_controller.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+from abc import abstractmethod
+from pathlib import Path
+from typing import Optional, Union
+
+from granulate_utils.linux.cgroups.base_controller import BaseController
+from granulate_utils.linux.cgroups.cgroup import CgroupCore, CgroupCoreV1, CgroupCoreV2
+
+
+class MemoryControllerInterface:
+    memory_limit: str
+    usage: str
+    max_usage_in_bytes: str = ""
+    swap_limit: str = ""
+
+    def __init__(self, cgroup: CgroupCore):
+        self.cgroup = cgroup
+
+    @abstractmethod
+    def set_limit_in_bytes(self, limit: int) -> None:
+        pass
+
+    def get_memory_limit(self) -> int:
+        return int(self.cgroup.read_from_interface_file(self.memory_limit))
+
+    def _set_swap_limit(self, limit: str) -> None:
+        try:
+            self.cgroup.write_to_interface_file(self.swap_limit, limit)
+        except PermissionError:
+            # if swap extension is not enabled (CONFIG_MEMCG_SWAP) this file doesn't exist
+            # and PermissionError is thrown (since it can't be created)
```

Comment:
Btw instead of this thing you could just fix `write_to_interface_file` to use a Python API which writes to EXISTING files without trying to create them. The equivalent of NOT passing `O_CREAT`.

Reply:
Then you catch the `FileNotFoundError` and code is clearer

Reply:
Resolving as this is no longer relevant